quinn-proto/src/connection/mod.rs

```diff
@@ -317,7 +317,10 @@ impl Connection {
             next_crypto: None,
             accepted_0rtt: false,
             permit_idle_reset: true,
-            idle_timeout: config.max_idle_timeout,
+            idle_timeout: config
+                .max_idle_timeout
+                .filter(|dur| dur.0 != 0)
+                .map(|dur| Duration::from_millis(dur.0)),
```

Comment:
Nit: instead of the `filter().map()` chaining, I think a `match` expression would make it easier to understand the logic here.